test/e2e/actuals_test.go

```diff
@@ -512,6 +512,119 @@ func customizedCRPStatusUpdatedActual(crpName string,
 	}
 }
 
+func safeDeploymentCRPStatusUpdatedActual(wantSelectedResourceIdentifiers []placementv1beta1.ResourceIdentifier, failedDeploymentResourceIdentifier placementv1beta1.ResourceIdentifier, wantSelectedClusters []string, wantObservedResourceIndex string) func() error {
+	return func() error {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+		crp := &placementv1beta1.ClusterResourcePlacement{}
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+			return err
+		}
+
+		var wantPlacementStatus []placementv1beta1.ResourcePlacementStatus
+		// We only expect the deployment to not be available on one cluster.
+		unavailableResourcePlacementStatus := placementv1beta1.ResourcePlacementStatus{
+			Conditions: []metav1.Condition{
+				{
+					Type:               string(placementv1beta1.ResourceScheduledConditionType),
+					Status:             metav1.ConditionTrue,
+					Reason:             condition.ScheduleSucceededReason,
+					ObservedGeneration: crp.Generation,
+				},
+				{
+					Type:               string(placementv1beta1.ResourceRolloutStartedConditionType),
+					Status:             metav1.ConditionTrue,
+					Reason:             condition.RolloutStartedReason,
+					ObservedGeneration: crp.Generation,
+				},
+				{
+					Type:               string(placementv1beta1.ResourceOverriddenConditionType),
+					Status:             metav1.ConditionTrue,
+					Reason:             condition.OverrideNotSpecifiedReason,
+					ObservedGeneration: crp.Generation,
+				},
+				{
+					Type:               string(placementv1beta1.ResourceWorkSynchronizedConditionType),
+					Status:             metav1.ConditionTrue,
+					Reason:             condition.AllWorkSyncedReason,
+					ObservedGeneration: crp.Generation,
+				},
+				{
+					Type:               string(placementv1beta1.ResourcesAppliedConditionType),
+					Status:             metav1.ConditionTrue,
+					Reason:             condition.AllWorkAppliedReason,
+					ObservedGeneration: crp.Generation,
+				},
+				{
+					Type:               string(placementv1beta1.ResourcesAvailableConditionType),
+					Status:             metav1.ConditionFalse,
+					Reason:             condition.WorkNotAvailableReason,
+					ObservedGeneration: crp.Generation,
+				},
+			},
+			FailedPlacements: []placementv1beta1.FailedResourcePlacement{
+				{
+					ResourceIdentifier: failedDeploymentResourceIdentifier,
+					Condition: metav1.Condition{
+						Type:               string(placementv1beta1.ResourcesAvailableConditionType),
+						Status:             metav1.ConditionFalse,
+						Reason:             "ManifestNotAvailableYet",
+						ObservedGeneration: crp.Generation + 1,
```

Comment:
there is actually no relationship between the crp generation and this failed manifest generation. This is the generation of the object that failed to apply/available